Add discover locations

morielnoam opened the pull request:
created the page and route for DiscoverLocations with design

morielnoam committed:
created initial version for discover locations

morielnoam committed:
designed the discover locations of page

danacohen111 commented on src/components/App.tsx:
```diff
@@ -12,6 +12,7 @@ const App = () => {
           <Route path="/signup" element={<SignupForm />} />
           <Route path="/ai-recommendations" element={<AIRecommendations />} />
           <Route path="/login" element={<LoginForm />} />
+          <Route path="/real-estate" element={<DiscoverLocations />} />
```
discover-locations

danacohen111 commented on src/styles/DiscoverLocations.css:
```diff
@@ -0,0 +1,135 @@
+.discover-locations-container {
```
fix the design, maybe remove the green background to create a cleaner look 
![image](https://github.com/user-attachments/assets/3132f30d-8b35-4c2e-916d-16cedd5719c7)

danacohen111 replied:
use react material lib as much as possible , look at the home page and the profiles components for reference

danacohen111 commented on src/components/DiscoverLocations/RealEstateModal.tsx:
```diff
@@ -0,0 +1,28 @@
+import React from "react";
```
put this file in a different folder named Realestate, and rename file to RealEstate

danacohen111 commented on src/components/DiscoverLocations/DiscoverLocations.tsx:
```diff
@@ -0,0 +1,100 @@
+    }, []);
+  
+    const handleCardClick = (realEstate: RealEstate) => {
+      setSelectedRealEstate(realEstate);
```
go to map when clicking discover button

morielnoam replied:
Added a button to click on when the user want to see the location of the real estate on the map.
At this point the button doing nothing, when the map page will be ready we will adjust the parameters and redirect accordingly.

danacohen111 commented on src/components/DiscoverLocations/RealEstateModal.tsx:
```diff
@@ -0,0 +1,28 @@
+  onClose: () => void;
+}
+
+const RealEstateModal: React.FC<RealEstateModalProps> = ({ realEstate, onClose }) => {
```
![image](https://github.com/user-attachments/assets/96ee2894-f5ac-4579-97bb-4baa6d993b9b)
 the links are over the card , and the design doesnt display the information well. Checkout the material lib to improve the design

morielnoam committed:
fixed the comments from the pr
reorder the modal files in matching folders
redesigned the discover-locations page
added button for each card to redirect to the map
changed the route from real-estates to discover-locations

morielnoam committed:
adjusted the response from using user-service.getUser to avoid conflicts

morielnoam committed:
Merge branch 'dev' into add_discover_locations

morielnoam committed:
Merge branch 'dev' into add_discover_locations
